app/src/org/commcare/android/logic/BarcodeScanListenerDefaultImpl.java

```diff
@@ -0,0 +1,121 @@
+package org.commcare.android.logic;
+
+import android.app.Activity;
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import org.commcare.dalvik.BuildConfig;
+import org.commcare.suite.model.Callout;
+import org.commcare.suite.model.CalloutData;
+
+import java.util.Hashtable;
+import java.util.Map;
+
+/**
+ * Created by dancluna on 8/5/15.
+ */
+public final class BarcodeScanListenerDefaultImpl {
+    public interface BarcodeScanListener {
+        void onBarcodeFetch(String result, Intent intent);
+
+        void onCalloutResult(String result, Intent intent);
+    }
+
+    public interface CalloutActionSetup {
+        void onImageFound(CalloutData calloutData);
+    }
+
+    public interface CalloutAction {
+        void callout();
+    }
+
+    public static final int BARCODE_FETCH = 1;
+    public static final int CALLOUT = 3;
+
+    public static void onBarcodeResult(BarcodeScanListener barcodeScanListener, int requestCode, int resultCode, Intent intent) {
+        if (BuildConfig.DEBUG) {
+            if (!(requestCode == BARCODE_FETCH)) {
```

Comment:
This is an assertion, only in a different style because AS complains that "assertions are not reliable" when I try to use an actual

```java
assert requestCode == BARCODE_FETCH
```